### ISSUE TITLE
arXiv '20 | TorchGPipe: On-the-fly Pipeline Parallelism for Training Giant Models

### ISSUE THREAD

**Issue by vycezhong** [vycezhong/read-papers#238]
https://arxiv.org/pdf/2004.09910.pdf

PyTorch的实现基于此. 好好研究.

**Comment by vycezhong** [comment]
这个图很有意思. GPipe + Gradient Checkpoint. 
![image](https://user-images.githubusercontent.com/25879526/130355357-9ce5357f-551b-4097-a91b-e53cd5495b03.png)

**Comment by vycezhong** [comment]
torchgpipe把模型看作是list of layers. 每个layer都是不可再分的，black-box，而且是`referentially transparent`的. 关于`referentially transparent`，wikipedia上给的解释如下:

> The expression value must be the same for the same inputs and its evaluation must have no side effects. 

看上去就是确定性的意思.

**Comment by vycezhong** [comment]
从Section 3.1开始是什么鬼，误入OSDI/SOSP片场？？？
